sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/OptimizerMaxIterationsSuite.scala

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.optimizer
+
+import org.apache.spark.sql.catalyst.dsl.plans._
+import org.apache.spark.sql.catalyst.errors.TreeNodeException
+import org.apache.spark.sql.catalyst.expressions.{Alias, IntegerLiteral, Literal}
+import org.apache.spark.sql.catalyst.plans.PlanTest
+import org.apache.spark.sql.catalyst.plans.logical.{LogicalPlan, OneRowRelation, Project}
+import org.apache.spark.sql.catalyst.rules.Rule
+import org.apache.spark.sql.internal.SQLConf
+
+/**
+ * A dummy optimizer rule for testing that decrements integer literals until 0.
+ */
+object DecrementLiterals extends Rule[LogicalPlan] {
+  def apply(plan: LogicalPlan): LogicalPlan = plan transformExpressions {
+    case IntegerLiteral(i) if i > 0 => Literal(i - 1)
+  }
+}
+
+class OptimizerMaxIterationsSuite extends PlanTest {
+  test("Adjust maxIterations") {
+    val iterations = 5
+    val maxIterationsNotEnough = 3
+    val maxIterationsEnough = 10
+    val analyzed = Project(Alias(Literal(iterations), "attr")() :: Nil, OneRowRelation()).analyze
+
+    conf.setConf(SQLConf.OPTIMIZER_MAX_ITERATIONS, maxIterationsNotEnough)
+    val optimizer = new SimpleTestOptimizer() {
```

Comment:
How about using `SparkOptimizer` instead of using `SimpleTestOptimizer`? That's because the existing test for `ANALYZER_MAX_ITERATIONS` uses `Analyzer` directly.

Reply:
It seems all other Optimizer-related tests are using `SimpleTestOptimizer`. I guess it is the proper one to be used based on the context.